test/functional/images_controller_test.rb

```diff
@@ -49,7 +49,7 @@ class ImagesControllerTest < ActionController::TestCase
     # This value is tested by the deprecation warning in application_controller
     # so we need to set it or the test will crash
     request.env['REQUEST_URI']="compute_resources/#{@image.compute_resource_id}/images"
-    get :index, { :format => :json, :compute_resource_id => @image.compute_resource_id }, set_session_user
+    get :index, { :format => :json, :compute_resource_id => @image.compute_resource_id, :operatingsystem_id => @image.operatingsystem_id, :architecture_id => @image.architecture_id }, set_session_user
```

Comment:
@iNecas It's not green if the OS parameter is removed - in that case it cannot find any images at all and the assertion in line 55 fails. I'm cool w/ rewriting this via factories but as it stands now it still tests the listing behavior well.